crypto/provider.c

```diff
@@ -39,6 +39,12 @@ OSSL_PROVIDER *OSSL_PROVIDER_try_load(OSSL_LIB_CTX *libctx, const char *name,
         ossl_provider_free(prov);
         return NULL;
     }
+    if (actual != prov) {
```

Comment:
There is one more place where ossl_provider_add_to_store() is called with actual possibly different (in provider_conf_load()). Might be the same change should be done there?